watcher.go

```diff
@@ -32,23 +35,52 @@ const (
 
 type Watcher interface {
 	GetQueries() []CarQuery
+	GetFlexCarQuery() CarQuery
 	OnCarAvailable(query CarQuery, cars []Car)
+	OnFlexCarAvailable(cars []FlexCarAvailabilityResp)
 }
 
 type WatcherOptions struct {
-	Interval time.Duration
-	Watcher  Watcher
+	Interval        time.Duration
+	Watcher         Watcher
+	IsFetchStations bool
+	IsFetchFlexCars bool
 }
 
-func StartWatcher(options WatcherOptions) {
-	checkForAvailabilities(options)
+func StartWatcher(ctx context.Context, options WatcherOptions) error {
```

Comment:
not sure `StartWatcher` should return if the communauto API returns a random error and recovers after some time

Bubbling up the errors to the caller through a callback could be a nice enhancement but 🤷